unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/Common/LoadingBar.cs

```diff
@@ -0,0 +1,24 @@
+using TMPro;
+using UnityEngine;
+
+[RequireComponent(typeof(RectTransform))]
+public class LoadingBar : MonoBehaviour
+{
+    public float currentPercentage { get; set; }
+
+    [SerializeField] internal TMP_Text percentageText;
+    [SerializeField] internal RectTransform loadingBar;
+
+    private float maxBarWeidth;
+
+    private void Awake() { maxBarWeidth = ((RectTransform)transform).rect.size.x; }
+
+    public void SetActive(bool isActive) { gameObject.SetActive(isActive); }
+
+    public void SetPercentage(float newValue)
+    {
+        currentPercentage = newValue;
+        percentageText.text = $"{newValue.ToString("0")}%";
+        loadingBar.sizeDelta = new Vector2(newValue * maxBarWeidth / 100f, loadingBar.sizeDelta.y);
```

Comment:
Shouldn't we override the pivot to be (0,0) ? or maybe it's not necessary

Reply:
In this case it's not necessary because the own loading bar game object is already configured to have the pivot in the position 0.